tests/runfiles/runfiles_test.py

```diff
@@ -525,6 +525,259 @@ def testDirectoryBasedRlocationWithRepoMappingFromOtherRepo(self) -> None:
+        }
+        prefixed_mappings = {
+            ("deps+", "external_dep"): "external_dep~prefix",
+            ("test_deps+", "test_lib"): "test_lib~2.1.0",
+        }
+        
+        return _RepositoryMapping(exact_mappings, prefixed_mappings)
+
+    def testRepositoryMappingLen(self) -> None:
+        """Test __len__ method - should count both exact and prefixed mappings."""
+        repo_mapping = self._createTestRepositoryMapping()
+        self.assertEqual(len(repo_mapping), 5)  # 3 exact + 2 prefixed
+        
+        # Test empty mapping
+        from python.runfiles.runfiles import _RepositoryMapping  # buildifier: disable=bzl-visibility
+        empty_mapping = _RepositoryMapping({}, {})
+        self.assertEqual(len(empty_mapping), 0)
+
+    def testRepositoryMappingGetItem(self) -> None:
+        """Test __getitem__ method - should work for both exact and prefix-based mappings."""
+        repo_mapping = self._createTestRepositoryMapping()
+        
+        # Test exact lookup
+        self.assertEqual(repo_mapping[("", "my_workspace")], "_main")
+        self.assertEqual(repo_mapping[("", "config_lib")], "config_lib~1.0.0")
+        self.assertEqual(repo_mapping[("deps+specific_repo", "external_dep")], "external_dep~exact")
+        
+        # Test prefix-based lookup
+        self.assertEqual(repo_mapping[("deps+some_repo", "external_dep")], "external_dep~prefix")
+        
+        # Test that exact takes precedence over prefix
+        self.assertEqual(repo_mapping[("deps+specific_repo", "external_dep")], "external_dep~exact")
+        
+        # Test KeyError for non-existent mapping (neither exact nor prefix match)
+        with self.assertRaises(KeyError):
+            _ = repo_mapping[("nonexistent", "repo")]
+
+    def testRepositoryMappingIter(self) -> None:
+        """Test __iter__ method - should iterate over all mapping keys (exact first, then prefixed)."""
+        repo_mapping = self._createTestRepositoryMapping()
+        
+        keys = list(repo_mapping)
+        self.assertEqual(len(keys), 5)  # 3 exact + 2 prefixed
+        
+        # Check that all expected keys are present
+        exact_keys = [("", "my_workspace"), ("", "config_lib"), ("deps+specific_repo", "external_dep")]
+        prefixed_keys = [("deps+", "external_dep"), ("test_deps+", "test_lib")]
+        
+        for exact_key in exact_keys:
+            self.assertIn(exact_key, keys)
+        
+        for prefixed_key in prefixed_keys:
+            self.assertIn(prefixed_key, keys)
+        
+        # Verify order: exact keys should come before prefixed keys
+        first_three_keys = keys[:3]
+        last_two_keys = keys[3:]
+        
+        for exact_key in exact_keys:
+            self.assertIn(exact_key, first_three_keys)
+        
+        for prefixed_key in prefixed_keys:
+            self.assertIn(prefixed_key, last_two_keys)
+
+    def testRepositoryMappingContains(self) -> None:
```

Comment:
Remove this test. The underlying class doesn't need to support contains checking.